packages/table/test/tableTests.tsx

```diff
@@ -56,6 +58,38 @@ describe("<Table>", () => {
         expect(table.find(".bp-table-column-headers .bp-table-header", 1).element).to.be.ok;
     });
 
+    it("Renders correctly with loading options", () => {
+        const renderCell = () => <Cell>my cell value</Cell>;
+        const loadingOptions = [
+            TableLoadingOption.CELLS,
+            TableLoadingOption.COLUMN_HEADERS,
+            TableLoadingOption.ROW_HEADERS,
+        ];
+        const tableHarness = harness.mount(
+            <Table loadingOptions={loadingOptions} numRows={2}>
+                <Column name="Column0" renderCell={renderCell} />
+                <Column name="Column1" renderCell={renderCell} />
+            </Table>,
+        );
+
+        expect(tableHarness.element.textContent).to.equal("");
+
+        const cells = tableHarness.element.querySelectorAll(".bp-table-cell");
+        for (let i = 0; i < cells.length; i++) {
+            expectCellLoading(cells.item(i), CellType.BODY_CELL);
+        }
+
+        const columnHeaders = tableHarness.element.querySelectorAll(".bp-table-column-headers .bp-table-header");
+        for (let i = 0; i < columnHeaders.length; i++) {
+            expectCellLoading(columnHeaders.item(i), CellType.COLUMN_HEADER);
+        }
+
+        const rowHeaders = tableHarness.element.querySelectorAll(".bp-table-row-headers .bp-table-header");
```

Comment:
you could use `.queryAll()` instead of `querySelectorAll()` (from `dom4` polyfill) which returns an Array, then simply do `rowHeaders.forEach(...)`